### PR TITLE
Fix cryo sleeping GC flood, stop snoring and apply stasis in cryo

### PR DESCRIPTION
**Pull request by Urumasi**
## About The Pull Request

Going SSD now uses a special type of sleep as to not conflict with normal sleeping, going SSD now uses perma sleep.
Going to sleep in cryo puts you in special cryosleep (removes SSD sleep) that doesn't make you snore.
Also before this the GC queue was being heavily spammed with sleep screen alert objects.

## Why It's Good For The Game

**Tim Torres** snores.
**Tim Torres** snores.
**Tim Torres** snores.
**Tim Torres** snores.
**Tim Torres** snores.
**Tim Torres** snores.

## Changelog
:cl:
tweak: Cryogenic storage now applies stasis to stop your metabolism.
tweak: You no longer snore while in deep cryogenic sleep.
/:cl:

### CODE REVIEW

**Comment by KubeRoot** [comment]
Inaccurate; every line of "**Tim Torres** snores." should have a 50% chance of having "(xN)" next to it, where N is a number from 2 to 4

File: code/modules/mob/living/carbon/life.dm [hunk 1]
```diff
@@ -21,9 +21,7 @@
 		if(.) //not dead
 			handle_blood()
 
-		if(isLivingSSD())//if you're disconnected, you're going to sleep
-			if(AmountSleeping() < 20)
-				AdjustSleeping(20)//adjust every 10 seconds
+		// SinguloStation edit (removal) - Fix SSD and cryo sleeping
```
Comment (KubeRoot) [inline code/modules/mob/living/carbon/life.dm]: You might want to handle removal by wrapping original code in block quotes.

File: code/modules/mob/living/status_procs.dm [hunk 2]
```diff
@@ -340,6 +340,63 @@
 		S = apply_status_effect(STATUS_EFFECT_SLEEPING, -1)
 	return S
 
+// SinguloStation edit begin - Fix SSD and cryo sleeping
```
Comment (KubeRoot) [inline code/modules/mob/living/status_procs.dm]: Those could go in a separate file in singulostation folder

Comment (KubeRoot) [reply code/modules/mob/living/status_procs.dm]: Also, should document what the difference from regular sleep is - presumably just no snoring

File: code/datums/status_effects/debuffs.dm [hunk 3]
```diff
@@ -174,6 +174,39 @@
...
+	alert_type = /obj/screen/alert/status_effect/cryosleep
+	needs_update_stat = TRUE
+
+/datum/status_effect/incapacitating/cryosleeping/on_apply()
+	. = ..()
+	if(!.)
+		return
+	var/datum/status_effect/incapacitating/sleeping/ssd/SSD = owner.IsSSDSleeping()
+	if(SSD)
+		SSD = null
```
Comment (KubeRoot) [inline code/datums/status_effects/debuffs.dm]: What's the purpose of setting the local `SSD` var to null? Wish it wasn't necessary to remove the other effect, but maybe they can't really coexist.

File: singulostation/code/datums/status_effects/misc_effects.dm [hunk 4]
```diff
@@ -16,7 +16,9 @@
 	alert_type = /obj/screen/alert/status_effect/cryo_protection
 
 /datum/status_effect/cryo_protection/on_apply()
-	owner.PermaSleeping()
+	owner.PermaCryoSleeping()
+	owner.apply_status_effect(STATUS_EFFECT_STASIS, STASIS_MACHINE_EFFECT)
```
Comment (KubeRoot) [inline singulostation/code/datums/status_effects/misc_effects.dm:20]: Is this exactly equivalent to stasis machines? A quick test on my local server shows that buckling myself to a stasis machine before SSDing (aghost) prevents snoring already.

File: code/datums/status_effects/debuffs.dm (hunk 3 above)
Comment (Urumasi) [reply code/datums/status_effects/debuffs.dm]: I'm not sure if this is a ref what would potentionally block it from qdeling (it qdels in the proc below) and I'm too afraid to think about it

**Review by MaltVinegar — APPROVED** [review]
Tested and it works

**Pull request merged**